conductor/src/main/java/com/bluelinelabs/conductor/RouterTransaction.kt

```diff
@@ -0,0 +1,112 @@
+package com.bluelinelabs.conductor
+
+import android.os.Bundle
+import androidx.annotation.RestrictTo
+import androidx.annotation.RestrictTo.Scope.LIBRARY
+import com.bluelinelabs.conductor.internal.TransactionIndexer
+
+private const val INVALID_INDEX = -1
+private const val KEY_VIEW_CONTROLLER_BUNDLE = "RouterTransaction.controller.bundle"
+private const val KEY_PUSH_TRANSITION = "RouterTransaction.pushControllerChangeHandler"
+private const val KEY_POP_TRANSITION = "RouterTransaction.popControllerChangeHandler"
+private const val KEY_TAG = "RouterTransaction.tag"
+private const val KEY_INDEX = "RouterTransaction.transactionIndex"
+private const val KEY_ATTACHED_TO_ROUTER = "RouterTransaction.attachedToRouter"
+
+/**
+ * Metadata used for adding [Controller]s to a [Router].
+ */
+class RouterTransaction
+private constructor(
+        @get:JvmName("controller")
+        val controller: Controller,
+        private var tag: String? = null,
+        private var pushControllerChangeHandler: ControllerChangeHandler? = null,
+        private var popControllerChangeHandler: ControllerChangeHandler? = null,
+        private var attachedToRouter: Boolean = false,
+        @RestrictTo(LIBRARY)
```

Comment:
can this be replaced by "internal" visibility? (not sure)

Reply:
Unfortunatelly not. Internal becomes public on java.